gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -539,6 +539,10 @@ OGRErr OGRFlatGeobufLayer::parseFeature(OGRFeature *poFeature, OGRGeometry **ogr
     if (VSIFReadL(&m_featureSize, sizeof(uoffset_t), 1, m_poFp) != 1)
         return CPLErrorIO();
     CPL_LSBPTR32(&m_featureSize);
+    if (m_featureSize > feature_max_buffer_size) {
+        CPLError(CE_Failure, CPLE_AppDefined, "Feature size too large (>= 2GB)");
```

Comment:
feature_max_buffer_size is actually 1 GB.

Reply:
Was a mistake, fixed.